subsys/mpsl/cx/thread/mpsl_cx_thread.c

```diff
@@ -82,27 +85,38 @@ static void gpiote_irq_handler(const struct device *gpiob, struct gpio_callback
 	(void)cb;
 	(void)pins;
 
-	static mpsl_cx_op_map_t last_notified;
 	int32_t ret;
 	mpsl_cx_op_map_t granted_ops;
-
-	if (callback != NULL) {
-		ret = granted_ops_get(&granted_ops);
-
-		__ASSERT(ret == 0, "Getting grant pin state returned unexpected result: %d", ret);
-		if (ret != 0) {
-			/* nrfx gpio implementation cannot return failure for this call
-			 * This condition is handled for fail-safe approach. It is assumed
-			 * GRANT is not given, if cannot read its value
-			 */
-			granted_ops = granted_ops_map(false);
+	uint8_t incremented = 0;
+
+	do {
+		if (callback != NULL && is_handled == 0) {
```

Comment:
This `if (callback)` inside the loop makes the code harder to follow. I think it would be better to extract it out of the loop with sth like 

```
if (callback == NULL) {
    return;
}
````